src/mon/PaxosService.h

```diff
@@ -852,6 +857,24 @@ class PaxosService {
     return cached_last_committed;
   }
 
+  /**
+   * Get the quantity of last to remove
+   *
+   * @returns Our last to remove quantity
+   */
+  version_t get_last_to_remove() const {
+    return last_to_remove;
+  }
+
+  /**
+   * Check if the store.db disk space is insufficient
+   *
+   * @returns true if insufficient space; false otherwise
+   * */
+  bool is_insufficient_space() const {
```

Comment:
We can return const ref here and avoid the additional initialization in `LogMonitor::preprocess_log`

Reply:
Either move to protected